hyper/http20/connection.py

```diff
@@ -34,6 +35,20 @@
 TRANSIENT_SSL_ERRORS = (ssl.SSL_ERROR_WANT_READ, ssl.SSL_ERROR_WANT_WRITE)
 
 
+class _LockedH2Connection(object):
+
+    def __init__(self, connection, lock):
+        self.lock = lock
+        self.connection = connection
+
+    def __enter__(self):
+        self.lock.acquire()
+        return self.connection
+
+    def __exit__(self, _exc_type, _exc_val, _exc_tb):
+        self.lock.release()
```

Comment:
Cool, this class looks good! Let's genericise it a bit though: there's actually nothing specific to the H2Connection about this class, so it can just be `LockedObject`. Additionally, we can move `connection` to `_obj` or something similar, just to indicate that you're really supposed to use the context manager to get it.